pkg/gofr/datasource/solr/solr_test.go

```diff
@@ -0,0 +1,178 @@
+package solr
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+)
+
+func Test_InvalidRequest(t *testing.T) {
+	_, err := call(context.TODO(), "GET", ":/localhost:", nil, nil)
+
+	require.Error(t, err, "TEST Failed.\n")
+}
+
+func Test_InvalidJSONBody(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write([]byte(`Not a JSON`))
+	}))
+	defer ts.Close()
+
+	_, err := call(context.TODO(), "GET", ts.URL, nil, nil)
+
+	require.Error(t, err, "TEST Failed.\n")
+}
+
+func Test_ErrorResponse(t *testing.T) {
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		http.Error(w, "some error", http.StatusLocked)
+	}))
+	ts.Close()
+
+	_, err := call(context.TODO(), "GET", ts.URL, nil, nil)
+
+	require.Error(t, err, "TEST Failed.\n")
+}
+
+func setupClient(t *testing.T) *Client {
+	t.Helper()
+
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write([]byte(`{
+  		"responseHeader": {
+    	"rf": 1,
+    	"status": 0}}`))
+	}))
```

Comment:
You removed the ts.Close from the code. Maybe because you face a problem with the scope of the defer.

Here you can simply use t.Cleanup


```suggestion
	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		_, _ = w.Write([]byte(`{
  		"responseHeader": {
    	"rf": 1,
    	"status": 0}}`))
	}))

	t.Cleanup(func() {
		ts.Close()
	})
```